src/main/scala/slack/api/SlackApiClient.scala

```diff
@@ -1,36 +1,59 @@
 package slack.api
 
-import slack.models._
-
 import java.io.File
-import scala.concurrent.duration._
-import scala.concurrent.Future
+import java.lang.reflect.{Field, Modifier}
+import java.net.InetSocketAddress
 
 import akka.actor.ActorSystem
-import akka.stream.ActorMaterializer
-import akka.stream.scaladsl.Source
-import akka.http.scaladsl.Http
 import akka.http.scaladsl.model._
 import akka.http.scaladsl.model.headers.{Authorization, OAuth2BearerToken}
+import akka.http.scaladsl.settings.{ClientConnectionSettings, ConnectionPoolSettings}
+import akka.http.scaladsl.{ClientTransport, Http}
 import akka.parboiled2.CharPredicate
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.StrictLogging
 import play.api.libs.json._
+import slack.models._
 
-object SlackApiClient {
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+
+object SlackApiClient extends StrictLogging {
+
+  private[this] val config   = ConfigFactory.load()
+  private[this] val useProxy = Try(config.getString("akka.http.client.useproxy")).fold(_ => false, _.toBoolean)
+
+  private[this] val maybeSettings: Option[ConnectionPoolSettings] = if (useProxy) {
+    val proxyHost = config.getString("akka.http.client.proxyHost")
```

Comment:
Why are we using `akka.http.` in the parameter name?
It somehow suggests this configuration key is processed by Akka Http library, while in fact it's not. It's processed by us (slack-scala-client) and fed into Akka Http.

Reply:
I'll change the names right away. 
What do you suggest instead of akka.http.client.useproxy akka.http.client.proxyHost akka.http.client.proxyPort ?